frame_test.go

```diff
@@ -131,3 +131,19 @@ func TestFrame(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, f6b, f7b)
 }
+
+func TestFrameMetadata(t *testing.T) {
```

Comment:
Can you move this logic at the bottom of `TestFrame` instead of creating a new `TestFrameMetadata`? You can rename `f` to `f8` in this case 👍